server/src/main/java/org/apache/cassandra/sidecar/config/yaml/DurationSpecImpl.java

```diff
@@ -0,0 +1,143 @@
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import org.apache.cassandra.sidecar.config.DurationSpec;
+import org.apache.cassandra.sidecar.exceptions.ConfigurationException;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Represents a positive time duration. Wrapper class for Cassandra Sidecar duration configuration parameters,
+ * providing to the users the opportunity to be able to provide configuration values with a unit of their choice
+ * in {@code sidecar.yaml} as per the available options. This class mirrors the Cassandra DurationSpec class.
+ */
+abstract class DurationSpecImpl implements DurationSpec
+{
+    /**
+     * The Regexp used to parse the duration provided as String.
+     */
+    private static final Pattern UNITS_PATTERN = Pattern.compile(("^(\\d+)(d|h|s|ms|m)$"));
+
+    private final long quantity;
+    private final TimeUnit unit;
+
+    DurationSpecImpl(String value)
+    {
+        Matcher matcher = UNITS_PATTERN.matcher(value);
+
+        if (matcher.find())
+        {
+            this.quantity = Long.parseLong(matcher.group(1));
+            this.unit = DurationSpec.fromSymbol(matcher.group(2));
+        }
+        else
+        {
+            throw new ConfigurationException(String.format("Invalid duration value %s. Only positive numbers with" +
+                                                           "the unit of %s are allowed", value, acceptedUnits(minimumUnit())));
+        }
+
+        validateMinUnit(unit, minimumUnit());
+        validateQuantity(value, quantity, unit, minimumUnit(), Long.MAX_VALUE);
+    }
+
+    public DurationSpecImpl(long quantity, TimeUnit unit)
+    {
+        this.quantity = quantity;
+        this.unit = unit;
+
+        validateMinUnit(unit, minimumUnit());
+    }
+
+    /**
+     * @return the minimum unit that can be represented by this type
+     */
+    public abstract TimeUnit minimumUnit();
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public long quantity()
+    {
+        return quantity;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public TimeUnit unit()
+    {
+        return unit;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String toString()
+    {
+        return "'" + quantity + "' " + unit.toString().toLowerCase();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public int compareTo(@NotNull DurationSpec that)
+    {
+        if (this.unit == that.unit())
+        {
+            return Long.compare(this.quantity, that.quantity());
+        }
+
+        TimeUnit minUnit = this.unit.compareTo(that.unit()) < 0 ? this.unit : that.unit();
+        return Long.compare(this.to(minUnit), that.to(minUnit));
+    }
+
+    void validateMinUnit(TimeUnit unit, TimeUnit minUnit)
+    {
+        if (unit.compareTo(minUnit) < 0)
+            throw new IllegalArgumentException(String.format("Invalid duration: %s Accepted units:%s",
```

Comment:
```suggestion
            throw new IllegalArgumentException(String.format("Invalid duration: %s Accepted units: %s",
```

Reply:
I am changing this anyway, I will push changes soon

Reply:
error messages have changed a bit, I made them very explicit on what the allowed values are